src/main/webapp/js/go-to-deal.js

```diff
@@ -319,6 +391,31 @@ function initDeal() {
       });
 }
 
+/**
+ * Handles restaurant selection
+ * @param {object} restaurant
+ */
+function selectRestaurant(restaurant) {
+  document.getElementById('restaurant-id-input').value = restaurant.id;
+  document.getElementById('restaurant-input').value = restaurant.name;
+}
+
+/**
+ * Checks if the dates of the form is ordered and displays error message.
+ * @return {boolean}
+ */
+function checkDatesOrdered() {
+  const start = document.getElementById('start-input');
+  const end = document.getElementById('end-input');
+  const message = document.getElementById('date-error-msg');
+  if (start.value && end.value && start.value > end.value) {
+    message.style.display = 'block';
+    return false;
+  }
+  message.style.display = 'none';
+  return true;
+}
+
```

Comment:
Should this logic be common with the page that adds deals as well? Is there other validation logic that could be made common too?